core/src/test/java/kafka/test/api/ShareConsumerTest.java

```diff
@@ -1459,6 +1464,234 @@ public void testLsoMovementByRecordsDeletion() {
         producer.close();
     }
 
+    @Test
+    public void testShareGroupStateTopicCreation() throws Exception {
+        Set<String> topics = listTopics().names().get();
+        // The __share_group_state topic is created on the first FIND_COORDINATOR RPC call for any share group topic partition.
+        // Since that does not happen automatically on cluster creation, we expect that the cluster will not have
+        // __share_group_state topic until yet.
+        assertFalse(topics.contains(SHARE_GROUP_STATE_TOPIC_NAME));
+
+        KafkaShareConsumer<byte[], byte[]> shareConsumer = createShareConsumer(new ByteArrayDeserializer(), new ByteArrayDeserializer(), "group1");
+        shareConsumer.subscribe(Collections.singleton(tp.topic()));
+
+        long duration = 45 * 1000; // 45 seconds
+
+        long startTime = System.currentTimeMillis();
+
+        // Poll for 45 seconds to ensure that the __share_group_state topic is created.
+        while (System.currentTimeMillis() - startTime < duration) {
```

Comment:
can we replace with `waitUntilTrue`

Reply:
Thanks for the review. Made the required changes in the latest commit.